### DISCUSSION TITLE
Why initializer for monocular camera only matches features at octave 0?

### DISCUSSION THREAD

**Discussion by mitul93** [stella-cv/stella_vslam#416] [Q&A] [answered]
Hi

I was tracing the code for monocular map initialization and found out that `area::match_in_consistent_area()` only compares the features of current frame and previous frame at octave 0.

https://github.com/stella-cv/stella_vslam/blob/main/src/stella_vslam/match/area.cc#L22

I tried to comment the check and the system initialized faster with much more initial key points.

Is there a reason to use only features at octave 0 during initialization?

**Comment by ymd-stella** [comment, marked as answer]
Initializer seems to be implemented under the assumption that octave 0 is used.
This can be found in the following line:

https://github.com/stella-cv/stella_vslam/blob/f94dbda3e6237f4a77f413bcf33282f4e91ac1c4/src/stella_vslam/initialize/perspective.cc#L52

`sigma` means standard deviation of measurement error.
When comparing features of different scales, appropriate sigma should be used from the table calculated below.

https://github.com/stella-cv/stella_vslam/blob/f94dbda3e6237f4a77f413bcf33282f4e91ac1c4/src/stella_vslam/feature/orb_params.cc#L53-L61

`initialize/bearing_vector.cc` does not use `sigma`, so if equirectangular images are used as input, the check may be removed.

**Reply by mitul93** [reply]
Thanks for explanation @ymd-stella
